### DISCUSSION TITLE
Growing MultilineTextInput on Android

### DISCUSSION THREAD

**Discussion by NadaComa** [beeware/toga#2615] [Q&A]
Hi all,

I have multiple widgets stacked up in COLUMN mode. One of them being a multiline textbox.

Is there a way to have this textbox grow in size when adding content? Ideally it should grow even beyond the visible space of the screen.

I imagine something similar to K9. When you start writing the content of your mail the size of the text box grows with each additional line you enter.

I tried using flex=1 but this only adjusts to the available space on screen. Once that is used up, you need to scroll inside the widget, which becomes unhandy for long text.

Thanks a lot

**Comment by freakboy3742** [comment]
In short - not really. 

You could add an `on_change` listener, and use that to modify the "height" attribute of the widget; but there's no mechanism to determine the "rendered height" of the text. You might be able to get an rendered size estimate by reaching down to the underlying toolkit, but at that point any code you're writing is platform specific.

A widget *cant'* grow being the visible space on the screen. This is why MultilineTextInput uses a scroll box - since the widget can contain an arbitrary amount of text, it is necessary to accomodate an arbitrary amount of text.

**Comment by NadaComa** [comment]
I knew it ... So will think about a different approach.

Side question: 
Playing around with the multilinetext widget on Android, scrolling behaves somewhat awkward. Any swipe is decelerated much faster than in other applications. So it feels like you have to force it to scroll. Does anyone have any idea where this is coming from?

**Reply by mhsmith** [reply]
I think I've noticed that before, but I don't know what the cause is.

**Comment by NadaComa** [comment]
Plus another thing I just realized:
- Put just a textbox in the app
- Fill it with text so it "overflows"
- Tab in the middle of the text so the keyboard opens
You can't scroll down to the end of the text. Maybe sometimes but you really need to try hard. 
You only see the end of the text with open keyboard if you close the keyboard, then tab into the textwidget at the end of the text which reopens the keyboard but puts it right under the textbox.

To make matters worse.
Adding a button below makes it even more cumbersome. Overall this can be fixed by putting everything into a scrollcontainer, but then scrolling within the textbox gets even worse. It kind of works as long as I keep to the edges of the screen.

Any idea, how I have to integrate the widget to get the same haptic experience I am used from other apps?

**Reply by mhsmith** [reply]
If you want help with this, please post the source code of an example app that demonstrates the problem. And also let us know which version of Toga you're using.

**Comment by NadaComa** [comment]
The following is based on Toga 0.4.4

```
class textbox(toga.App):
    def startup(self):
        box = toga.Box(style=Pack(direction=COLUMN,flex=1))

        textbox = toga.MultilineTextInput(style=Pack(flex=1))
        button = toga.Button("Button", style=Pack(padding=5,flex=1))
        button2 = toga.Button("Button2", style=Pack(padding=5,flex=1))
        box.add(button)        
        box.add(textbox)
        box.add(button2)

        scrollcontainer = toga.ScrollContainer(content=box)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = scrollcontainer
        self.main_window.show()


def main():
    return textbox()
```

**Comment by NadaComa** [comment]
Hi all,
would anyone be able to have a look at this? Any help would be highly appreciated.
I really like this framework but large amounts of text are just not scrollable with the code above.

Cheers,
Nada

**Reply by mhsmith** [reply]
Sure, I'll take a look at it just now.

**Reply by mhsmith** [reply]
It looks like there's a bug here: let's continue at #2634.
